### PR TITLE
Bump `@metamask/snaps-*` packages

### PR DESCRIPTION
**Pull request by GuillaumeRx**
This bumps the snaps packages to the latest version.

Adds ES2020 functionalities by target it.

Also removes the private field from the subpackages `package.json`.

Fixes: #60 #61

### CODE REVIEW

**Comment by GuillaumeRx** [comment]
Waiting on a new release of `@metamask/snaps-cli`

**Review by FrederikBolding — COMMENTED** [review]
Can we also make it clear that it requires at least Node v18.6.0 or v16.17.0? Maybe tweaking the .nvmrc?

File: packages/site/package.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "site",
   "version": "0.1.0",
-  "private": true,
```
Comment (FrederikBolding) [inline packages/site/package.json:4]: This should remain private

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v16
+v18.6.0
```
Comment (FrederikBolding) [inline .nvmrc:1]: Is there a way to allow >=18.6.0? 🤔

Comment (GuillaumeRx) [reply .nvmrc:1]: There's not much doc about `.nvmrc`. All that I've found was fixed versions

Comment (GuillaumeRx) [reply .nvmrc:1]: I added the enforcement of `>=18.6.0` in `package.json` in a75f4ec but there is no way to specify a version range for `.nvmrc`

File: packages/snap/package.json [hunk 3]
```diff
@@ -39,8 +38,8 @@
     "@metamask/eslint-config-jest": "^10.0.0",
     "@metamask/eslint-config-nodejs": "^10.0.0",
     "@metamask/eslint-config-typescript": "^10.0.0",
-    "@metamask/snaps-cli": "^0.38.2-flask.1",
-    "@metamask/snaps-jest": "^0.37.3-flask.1",
+    "@metamask/snaps-cli": "^2.0.0",
```
Comment (FrederikBolding) [inline packages/snap/package.json:41]:
```suggestion
    "@metamask/snaps-cli": "^2.0.1",
```

File: .nvmrc (hunk 2 above)
Comment (FrederikBolding) [inline .nvmrc:1]:
```suggestion
v18.6
```
I think we can allow patches?

File: yarn.lock [hunk 4]
```diff
@@ -4595,122 +4475,44 @@ __metadata:
...
-    validate-npm-package-name: ^5.0.0
-  checksum: eeb31013a8af39b5488fdb8a6a72e11c54274e423e0f2560fd5cf4ea0278e278aff60be070c1e5b847d84220f15338e01c4725ce7ebb5f3fc614553ae4235f44
+  checksum: 0a8886402aea3aeebc18dd8058f169b7c751f93aac22286a693dbb6f787b10e44f7da9efa40981dde427f2301ce3da11af93e37398157eb75fdce85c53f48dd8
   languageName: node
   linkType: hard
 
-"@metamask/snaps-utils@npm:^0.38.0-flask.1":
-  version: 0.38.0-flask.1
-  resolution: "@metamask/snaps-utils@npm:0.38.0-flask.1"
+"@metamask/snaps-utils@npm:^2.0.0":
```
Comment (FrederikBolding) [inline yarn.lock:4504]: Dedupe please

File: tsconfig.json [hunk 5]
```diff
@@ -3,12 +3,13 @@
...
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
+    "lib": ["DOM", "ES2020"],
     "module": "CommonJS",
     "moduleResolution": "node",
     "useUnknownInCatchVariables": false,
     "sourceMap": true,
     "strict": true,
-    "target": "ES2017"
+    "target": "ES2020"
```
Comment (david0xd) [inline tsconfig.json:12]: Can you leave a comment in the PR description that this is changed and why?

Comment (GuillaumeRx) [reply tsconfig.json:12]: This is purely to align with `snaps-monorepo` and unlock ES2020 functionalities

Comment (david0xd) [reply tsconfig.json:12]: Ok, thanks!

**Review by david0xd — APPROVED** [review]

**Pull request merged**